system-tests/lib/cre/capabilities/writesolana/write_solana.go

```diff
@@ -47,3 +61,186 @@ func registerWithV1(_ []string, nodeSetInput *cre.CapabilitiesAwareNodeSet) ([]k
 
 	return capabilities, nil
 }
+
+func transformNodeConfig(input cre.GenerateConfigsInput, existingConfigs cre.NodeIndexToConfigOverride) (cre.NodeIndexToConfigOverride, error) {
+	if !flags.HasFlagForAnyChain(input.NodeSet.ComputedCapabilities, flag) {
+		return existingConfigs, nil
+	}
+
+	data := solanaInput{}
+	for _, bcOut := range input.BlockchainOutput {
+		if bcOut.SolChain == nil {
+			continue
+		}
+		data.ChainSelector = bcOut.SolChain.ChainSelector
+		// find Solana forwarder address
+		forwarders := input.Datastore.Addresses().Filter(datastore.AddressRefByChainSelector(data.ChainSelector))
+		for _, addr := range forwarders {
+			if addr.Type == ks_sol.ForwarderState {
+				data.ForwarderState = addr.Address
+				continue
+			}
+			data.ForwarderAddress = addr.Address
+		}
+
+		break
+	}
+
+	workflowNodeSet, err := node.FindManyWithLabel(input.DonMetadata.NodesMetadata, &cre.Label{Key: node.NodeTypeKey, Value: cre.WorkerNode}, node.EqualLabels)
+	if err != nil {
+		return nil, errors.Wrap(err, "failed to find worker nodes")
+	}
+
+	for i := range workflowNodeSet {
+		var nodeIndex int
+		for _, label := range workflowNodeSet[i].Labels {
+			if label.Key == node.IndexKey {
+				nodeIndex, err = strconv.Atoi(label.Value)
+				if err != nil {
+					return nil, errors.Wrap(err, "failed to convert node index to int")
+				}
+			}
+		}
+
+		// find node's Solana address
+		expectedAddressKey := node.AddressKeyFromSelector(data.ChainSelector)
+		for _, label := range workflowNodeSet[i].Labels {
```

Comment:
Note for a future refactor: I think I count two instances in this function of reading from labels; ideally we'd transform those labels into a NodeInfo struct or similar that describes the node.

That alone should simplify this function quite a bit I would think